### PR TITLE
Survive after fork()

### PR DESCRIPTION
**Pull request by vmarkovtsev**
Since fork() kills all the side threads and makes processes share the same
file descriptors, there are two problems:

1. Worker's thread silently stops to exist
2. Client's socket is used concurrently by several processes which lead to
all kinds of errors, such as invalid SSL version, etc.

Since there is no fork() on Windows, the best way to check if we need to
reinitialize everything is to check if the worker's thread still runs.

Please note: SyncTransport is left intact and thus it is still vulnerable.
I suggest to fix it in the future PR because it will require PID tracking, etc.

### CODE REVIEW

File: gcloud/logging/handlers/transports/background_thread.py [hunk 1]
```diff
@@ -59,6 +59,10 @@ def __init__(self, logger):
 
         self._start()
 
+    @property
+    def alive(self):
+        return self._thread.is_alive()
```
Comment (dhermes) [inline gcloud/logging/handlers/transports/background_thread.py]: Docstring and unit-test please

File: gcloud/logging/handlers/transports/background_thread.py [hunk 2]
```diff
@@ -160,4 +164,6 @@ def send(self, record, message):
         :type message: str
         :param message: The formatted log message
         """
+        if not self.worker.alive:
+            self.__init__(self.client, self.worker.logger.name)
```
Comment (dhermes) [inline gcloud/logging/handlers/transports/background_thread.py]: `self.__init__` shouldn't be called, there must be a better way to do this

Comment (vmarkovtsev) [inline gcloud/logging/handlers/transports/background_thread.py]: Why?
Do you understand the problem well? It must be exactly __init__.

**Comment by vmarkovtsev** [comment]
I would note that this issue blocks the whole multiprocessing package, limiting users to the single process. And we have to initialize the transport from scratch, calling \__init__ as if we construct it from scratch. It is a common fork() survival pattern.

**Comment by dhermes** [comment]
@vmarkovtsev @waprin Is this going to be revived on `master`?

**Pull request closed without merging**